### ISSUE TITLE
SiriProxy on Ubuntu with iPhone 5 + 6.1 issues

### ISSUE THREAD

**Issue by ClemsonJeeper** [plamoni/SiriProxy#455]
So I have SiriProxy up and running and definitely have all my DNS sorted out (when I ping guzzoni.apple.com from my iphone, it resolves to my SiriProxy box).

hitman@media:~/SiriProxy$ rvmsudo siriproxy server
Starting SiriProxy on 0.0.0.0:443..
SiriProxy up and running.

At this point I am ssh'd into my iPhone and 'telnet guzzoni.apple.com 443' and SiriProxy does:

Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002cacec8 @signature=3, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002cacd60>, @zip_stream=#Zlib::Deflate:0x00000002cacd38, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins loaded: [#<SiriProxy::Plugin::Example:0x000000026f7c80 @manager=#<SiriProxy::PluginManager:0x00000002cac7c0 @plugins=[...]>>

Which is what I would expect.

However, when I go into Siri itself and "test siri proxy", it responds with a generic Siri message (I'm completely operational, and all my circuits...)

While on the server:

Starting SiriProxy on 0.0.0.0:443..
SiriProxy up and running.
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002c98090 @signature=3, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002c97f28>, @zip_stream=#Zlib::Deflate:0x00000002c97f00, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins loaded: [#<SiriProxy::Plugin::Example:0x00000002ab1718 @manager=#<SiriProxy::PluginManager:0x00000002c97988 @plugins=[...]>>]
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002ab15d8 @signature=4, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002ab1448>, @zip_stream=#Zlib::Deflate:0x00000002ab1420, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins loaded: [#<SiriProxy::Plugin::Example:0x00000002ab09f8 @manager=#<SiriProxy::PluginManager:0x00000002ab1060 @plugins=[...]>>]
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002ab0840 @signature=5, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002ab06b0>, @zip_stream=#Zlib::Deflate:0x00000002ab0688, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins loaded: [#<SiriProxy::Plugin::Example:0x00000002aadf00 @manager=#<SiriProxy::PluginManager:0x00000002aabef8 @plugins=[...]>>]
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002aae680 @signature=6, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002aaf210>, @zip_stream=#Zlib::Deflate:0x00000002aaf328, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins loaded: [#<SiriProxy::Plugin::Example:0x00000002a1d9a0 @manager=#<SiriProxy::PluginManager:0x000000027d91a8 @plugins=[...]>>]
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002a9add8 @signature=7, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002a9d858>, @zip_stream=#Zlib::Deflate:0x00000002a9da88, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins loaded: [#<SiriProxy::Plugin::Example:0x00000002c77548 @manager=#<SiriProxy::PluginManager:0x00000002ca1820 @plugins=[...]>>]
Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x00000002c74690 @signature=8, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x00000002c71be8>, @zip_stream=#Zlib::Deflate:0x00000002c6a4b0, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins loaded: [#<SiriProxy::Plugin::Example:0x00000002411f20 @manager=#<SiriProxy::PluginManager:0x000000026c0500 @plugins=[...]>>]

What am I missing?

**Comment by Deanmv** [comment]
`Starting SiriProxy on 0.0.0.0:443..`

You haven't set an IP in the config.yml file.

**Comment by ClemsonJeeper** [comment]
0.0.0.0 binds on all IP addresses.

I closed the issue - my issue turned out to be that dnsmasq was injecting 127.0.0.1 into /etc/resolv.conf on the dnsmasq machine.  So lookups on the dnsmasq machine for guzzoni.apple.com were routed back to itself.

I manually fixed /etc/resolv.conf to point to my correct name server instead of the symlink to the resolvconf runtime file, and everything magically started working (even with the 0.0.0.0:443 binding).
